### PR TITLE
Minor changes in Chapter 12

### PR DESCRIPTION
**Pull request by pavpanchekha**
These are mostly small wording or code style things, but it also fixes an issue that could occur if you had multiple `setTimeout` requests that fired in a different order from how they were created.

### CODE REVIEW

File: book/scheduling.md [hunk 1]
```diff
@@ -383,7 +384,7 @@ do security checks:
 
 ``` {.python}
 class JSContext:
-    def XMLHttpRequest_send(self, method, url, body, is_async, handle):
+    def XMLHttpRequest_send(self, method, url, body, isasync, handle):
```
Comment (chrishtr) [inline book/scheduling.md:394]: Why this change?

**Comment by chrishtr** [comment]
Which line of code fixes the bug you mentioned?

**Comment by pavpanchekha** [comment]
The line that fixes the bug is in the JavaScript, where the line `delete XHR_REQUESTS[handle]` would remove a key, which would potentially make the handle assigned to the next XHR be a handle that had already been assigned. By removing the `delete` line I've introduced a potential memory leak, but we already had memory leaks (and they're called out in the JS chapter) so I think it's OK. (Alternatively, we could fix it by introducing a real counter... I'll at least throw up a footnote with a warning.)

**Comment by pavpanchekha** [comment]
Added a footnote earlier, when discussing `setTimeout`, since it has a similar issue.

**Comment by chrishtr** [comment]
> The line that fixes the bug is in the JavaScript, where the line `delete XHR_REQUESTS[handle]` would remove a key, which would potentially make the handle assigned to the next XHR be a handle that had already been assigned. By removing the `delete` line I've introduced a potential memory leak, but we already had memory leaks (and they're called out in the JS chapter) so I think it's OK. (Alternatively, we could fix it by introducing a real counter... I'll at least throw up a footnote with a warning.)

Does the javascript runtime need to be in this PR then?

**Comment by pavpanchekha** [comment]
Please see latest commit message :)

**Review by chrishtr — APPROVED** [review]

File: book/scheduling.md (hunk 1 above)
Comment (pavpanchekha) [reply book/scheduling.md:394]: lol, these code blocks were one character too wide, and I didn't want to spill the arguments over two lines.
